File: backend/src/main/java/com/coril/mutualfund/domain/Movement.java
package com.coril.mutualfund.domain;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "movements")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Movement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String investorId;

    @Column(nullable = false)
    private String fundName;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OperationType operationType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MovementStatus status;

    @Column(nullable = false, precision = 18, scale = 2)
    private BigDecimal amount;

    @Column(nullable = false)
    private LocalDate date;
}

File: frontend/src/types/movement.ts
export type OperationType = 'SUBSCRIPTION' | 'REDEMPTION';

export type MovementStatus = 'PENDING' | 'COMPLETED' | 'REJECTED';

export interface Movement {
  id: number;
  fundName: string;
  operationType: OperationType;
  status: MovementStatus;
  amount: number;
  date: string; // ISO date string: "YYYY-MM-DD"
}
